### PR TITLE
FEATURE: MSF-20653 Support Blob Storage input source

### PR DESCRIPTION
**Pull request by tqtu**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/757/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/758/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/759/ : SUCCESS in 11s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+module GoodData
+  class BlobStorageClient
+    SAS_URL_PATTERN = %r{(^https?:\/\/[^\/]*)\/.*\?(.*)}
+    attr_reader :use_sas
+
+    def initialize(options = {})
+      raise("Data Source needs a client to Blob Storage to be able to get blob file but 'blobStorage_client' is empty.") unless options['blobStorage_client']
+
+      # rubocop:disable Naming/VariableName
+      if options['blobStorage_client']['connectionString'] || options['blobStorage_client']['container']
```
Comment (phong-nguyen-duy) [inline lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:21]: Both connectionString and container are mandatory

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/760/ : SUCCESS in 6s

File: lib/gooddata/helpers/data_helper.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@
 require 'csv'
 require 'digest'
 require 'open-uri'
+require_relative '../cloud_resources/blobstorage/blobstorage_client'
```
Comment (phong-nguyen-duy) [inline lib/gooddata/helpers/data_helper.rb:10]: should load blob storage api only when user defined blob storage input source

**Comment by phong-nguyen-duy** [comment]
@tqtu please add integration test to cover the new input source type

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb (hunk 1 above)
Comment (tqtu) [reply lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:20]: Fixed, connectionString and container are mandatory

File: lib/gooddata/helpers/data_helper.rb (hunk 2 above)
Comment (tqtu) [reply lib/gooddata/helpers/data_helper.rb:10]: Fixed, Moved require_relative into blobStorage

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/761/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/762/ : SUCCESS in 8s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      else
+        raise('Missing connection info for Blob Storage client')
+      end
+    end
+
+    def realize_blob(file, _params)
+      GoodData.gd_logger.info("Realizing download from Blob Storage. Container #{@container}.")
+      connect
+      filename = "#{SecureRandom.urlsafe_base64(6)}_#{Time.now.to_i}.csv"
+      blob_name = @path ? "#{file}" : "#{@path}/#{file}"
```
Comment (phong-nguyen-duy) [inline lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:35]: should handle the case @path ends with slash (/)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/763/ : SUCCESS in 10s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb (hunk 3 above)
Comment (tqtu) [reply lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb]: The path added delete_suffix to remove slash

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1784

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/560/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/20/ : SUCCESS in 1m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/105/ : FAILURE in 2m 59s (non-voting)

**Pull request merged**